c10/xpu/test/impl/XPUDeviceTest.cpp

```diff
@@ -0,0 +1,59 @@
+#include <gtest/gtest.h>
+
+#include <c10/xpu/XPUFunctions.h>
+
+#define ASSERT_EQ_XPU(X, Y) \
+  {                         \
+    bool _isEQ = X == Y;    \
+    ASSERT_TRUE(_isEQ);     \
+  }
+
+bool has_xpu() {
+  return c10::xpu::device_count() > 0;
+}
+
+TEST(XPUDeviceTest, DeviceBehavior) {
+  if (!has_xpu()) {
+    return;
+  }
+
+  c10::xpu::set_device(0);
+  ASSERT_EQ_XPU(c10::xpu::current_device(), 0);
+
+  if (c10::xpu::device_count() <= 1) {
```

Comment:
If device_count is 1, then changing device should not work, should it?
```suggestion
  if (c10::xpu::device_count() <= 2) {
```

Reply:
yes, if `device_count` is 1, it should return early.